lib/sedna/algorithms/multi_task_learning/multi_task_learning.py

```diff
@@ -0,0 +1,296 @@
+
+        if not method_name:
+            task_definition = self.method_selection.get("task_definition",
+                                                        "TaskDefinitionByDataAttr")
+            method_name = self._method_pair.get(task_definition,
+                                                'TaskMiningByDataAttr')
+            extend_param = self.parse_param(self.method_selection.get("task_definition_param"))
+        method_cls = ClassFactory.get_cls(ClassType.MTL, method_name)(
+            task_extractor=self.extractor, **extend_param
+        )
+        return method_cls(samples=samples)
+
+    def task_remodeling(self, samples, mappings):
+        method_name = self.method_selection.get("task_remodeling")
+        extend_param = self.parse_param(self.method_selection.get("task_remodeling_param"))
+        method_cls = ClassFactory.get_cls(ClassType.MTL, method_name)(models=self.models, **extend_param)
+        return method_cls(samples=samples, mappings=mappings)
+
+    def inference_integrate(self, tasks):
+        method_name = self.method_selection.get("inference_integrate")
+        extend_param = self.parse_param(self.method_selection.get("inference_integrate_param"))
+        method_cls = ClassFactory.get_cls(ClassType.MTL, method_name)(models=self.models, **extend_param)
+        return method_cls(tasks=tasks) if method_cls else tasks
+
+    def train(self, train_data: BaseDataSource,
+              valid_data: BaseDataSource = None,
+              post_process=None, **kwargs):
+        tasks, task_extractor, train_data = self.task_definition(train_data)
+        self.extractor = task_extractor
+        task_groups = self.task_relationship_discovery(tasks)
+        self.models = []
+        callback = None
+        if post_process:
+            callback = ClassFactory.get_cls(ClassType.CALLBACK, post_process)()
+        self.task_groups = []
+        feedback = {}
+        rare_task = []
+        for i, task in enumerate(task_groups):
+            if not isinstance(task, TaskGroup):
+                rare_task.append(i)
+                self.models.append(None)
+                self.task_groups.append(None)
+                continue
+            if not (task.samples and len(task.samples) > self.min_train_sample):
+                self.models.append(None)
+                self.task_groups.append(None)
+                rare_task.append(i)
+                n = len(task.samples)
+                sednaLogger.info(f"MTL Train Fail because of num  {n}  too least: {task.entry}")
+                continue
+            sednaLogger.info(f"MTL Train start {i} : {task.entry}")
+
+            model_obj = set_backend(estimator=self.base_model)
+            res = model_obj.train(train_data=task.samples, **kwargs)
+            if callback:
+                res = callback(model_obj, res)
+            model_path = model_obj.save(model_name=f"{task.entry}.pkl")
```

Comment:
why the model end with fix string "pkl",  other model format should be considered.

Reply:
fix